### ISSUE TITLE
[Suggestion] better Arrowdogde

### ISSUE THREAD

**Issue by Zeddesnetos** [MeteorDevelopment/meteor-client#1120]
**Describe the feature**
Better Arrowdogde

**Additional info**
The Arrowdodge from Meteor only works about 30% of the time.
ArrowDogde doesnt works when the arrow is fired when the "shooter" is near me, like about 6-7 blocks or less.
It does evade shoots almost always that would not hit me anyways, and the shots that would hit me only about 30% of the time the evading works.
ArrowDogde allways fails to evade a shot if it is fired with an crossbow with multishot,
Why does it try to dogde Potions? it doesnt work anyway cause potions have splash damage.
Sometimes while its trying to evade a shot, moves in the direction the shot is flying.
When near blocks, the evading fails. Why not jump over the blocks or just evade in the free direktion?

PS: im not very good in English, you can if you want formulate it  better, and post it as answer.

**Comment by unhookedhook** [comment]
I agree. Right now arrowdodge is a very basic module that just blindly leaps in a random direction when it detects any projectile. I would suggest 
NOT dodging the players own projectiles
Checking for a clear area to dodge in to
Check for walls/blocks
Check for LAVA (right now arrowdodge will JUMP INTO LAVA to avoid an arrow)
Calculate the trajectory of a projectile, and then move out of it's path.

**Comment by DesiCow** [comment]
Added in #2111
